### ISSUE TITLE
Trouble with FP (r7) on ARM

### ISSUE THREAD

**Issue by xarkes** [wangbj/syscalls#40]
Hello,
The project mentions ARM is supported, however I get the following error when trying to compile as simple call to the syscall wrapper:

```
$ cargo build --target=armv7-linux-androideabi
...
error: cannot use register `r7`: the frame pointer (r7) cannot be used as an operand for inline asm
  --> $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/syscalls-0.6.12/src/syscall/arm.rs:29:9
   |
29 |         in("r7") n as usize,
   |         ^^^^^^^^^^^^^^^^^^^
```

I am using the following version of Rust
```
cargo --version
cargo 1.70.0 (ec8a8a0ca 2023-04-25)
```
Note that I have no error when compiling for aarch64

Let me know if you need more details.

**Comment by jasonwhite** [comment]
Thanks for the bug report! Should be fixed in `v0.6.13`.

**Comment by xarkes** [comment]
Awesome, thanks!
